src/Chapter4.hs

```diff
@@ -672,7 +693,25 @@ Specifically,
    subtree of a tree
  ❃ Implement the function to convert Tree to list
 -}
+data Tree a = Leaf | Branch a (Tree a) (Tree a) deriving (Show)
+
+instance Functor Tree where
+  fmap _ Leaf = Leaf
+  fmap f (Branch a l r) = Branch (f a) (fmap f l) (fmap f r) 
+
+reverseTree :: Tree a -> Tree a
+reverseTree Leaf = Leaf
+reverseTree (Branch a l r) = Branch a (reverseTree r) (reverseTree l)
+
+
+treeToList :: Tree a -> [a] 
+treeToList Leaf = []
+treeToList (Branch a l r) = a : (treeToList l ++ treeToList r)
 
+insert :: Ord a => a -> Tree a -> Tree a 
+insert a Leaf = Branch a Leaf Leaf
+insert a (Branch b l r) | a > b = Branch b (insert a l) r
+insert a (Branch b l r) = Branch b l (insert a r)
```

Comment:
That wasn't necessary, but nice job on implementing the insert function for Binary Search Tree! Usually, it's better implemented using the `compare` function:

```haskell
insert a (Branch b l r) = case compare a b of
    EQ -> ...
    LT -> ...
    GT -> ...
```

so you can handle all 3 cases of comparison explicitly and more visible